src/sentry/processing/realtime_metrics/redis.py

```diff
@@ -84,15 +84,14 @@ def increment_project_duration_counter(
         Calling this increments the counter of the current time-window bucket with "timestamp" providing
         the time of the event in seconds since the UNIX epoch and "duration" the processing time in seconds.
         """
-        if self._histogram_bucket_size > 1:
-            timestamp -= timestamp % self._histogram_bucket_size
+        timestamp -= timestamp % self._histogram_bucket_size
```

Comment:
why the removal of the check here? did you end up finding out when and where `validate` is invoked to prevent `self._histogram_bucket_size` from being set to 0?

Reply:
Well, if `bucket_size == 1` then `timestamp % bucket_size == 0` and `timestamp - timestamp % bucket_size == timestamp`, so it behaves correctly in that case. Not sure what I was thinking when I added that check.

Reply:
i suppose i'm worried about `bucket_size == 0`, since that triggers throws a mod by zero error here, and in other places. we could omit checks on the bucket sizes if we can guarantee that it isn't possible to initialize a store with 0-sized buckets.